### DISCUSSION TITLE
App window opens on macOS startup

### DISCUSSION THREAD

**Discussion by GavBeats** [noah-nuebling/mac-mouse-fix#1177] [Questions] [answered]
When I login to my mac from a cold start, the mouse fix app window opens when the app launches at login. Not a huge deal, but kind of annoying to have to close it all the time. Is there a setting for this, or is it a bug?

Thanks.

**Comment by noah-nuebling** [comment]
Hey, @GavBeats, thanks for reaching out. 

If the app opens up automatically on startup, that's probably because it's set as a "Login Item" in macOS. See [Apple's Guide](https://support.apple.com/en-me/guide/mac-help/mh15189/mac) on how to manage those.

I hope that solves the problem, if not, let me know.

**Comment by GavBeats** [comment]
Hi, I’m fine with it opening on startup it’s just that the windows always appears on startup. I was wondering if there was a way to have it start on startup but in the background and not have the window appear every time.
Thanks.
________________________________

**Reply by noah-nuebling** [reply, marked as answer]
Hey @GavBeats, you can remove the app as a Login Item and it will stay active, even after closing the app or restarting the computer. 
As long as the 'Enable Mac Mouse Fix' switch is turned on, the 'Mac Mouse Fix Helper' app will run in the background and continue to affect how your mouse works – regardless of whether the main 'Mac Mouse Fix' app is running. 

You're not the only one to be confused by this, I think many people expect the app to work differently because most other apps don't work this way.

Originally, Mac Mouse Fix (or just 'Mouse Fix' back then) was a System Preferences plugin (prefpane) – For those, this behavior seems quite intuitive. But it became much harder to develop prefpanes and so I turned MMF into a normal app. The way the app is enabled was very deeply ingrained into the app's architecture at this point and it also has some benefits, so I decided to keep that – much to the confusion of some users.

In Mac Mouse Fix 3 I added text under the "Enable Mac Mouse Fix" button that says \
"Mac Mouse Fix will stay enabled after you close it"

That was meant to help with this misunderstanding. It did seem to help quite a bit but I still get such questions once in a while. So if you have ideas on how to communicate this a bit more clearly in the app or elsewhere, I'd be highly interested to hear about that!
